rust/src/delta_arrow.rs

```diff
@@ -732,4 +732,63 @@ mod tests {
             crate::SchemaDataType::primitive("timestamp".to_string())
         );
     }
+
+    #[test]
+    fn test_max_min_schema_for_fields_simple_type() {
+        let mut dest: Vec<ArrowField> = Vec::new();
+        let f = ArrowField::new("simple", ArrowDataType::Int32, true);
+        let expected = vec![ArrowField::new("simple", ArrowDataType::Int32, true)];
+        max_min_schema_for_fields(&mut dest, &f);
+        assert_eq!(dest, expected)
+    }
+
+    #[test]
+    fn test_max_min_schema_for_fields_struct_type() {
+        let mut dest: Vec<ArrowField> = Vec::new();
+        let f = ArrowField::new(
+            "struct",
+            ArrowDataType::Struct(vec![ArrowField::new("simple", ArrowDataType::Int32, true)]),
+            true,
+        );
+        let expected = vec![ArrowField::new(
+            "struct",
+            ArrowDataType::Struct(vec![ArrowField::new("simple", ArrowDataType::Int32, true)]),
+            true,
+        )];
```

Comment:
We don't have to repeat this, and it's probably clearer they are meant to be the same field if we just clone `f`. Same applies for test above.
```suggestion
        let f = ArrowField::new(
            "struct",
            ArrowDataType::Struct(vec![ArrowField::new("simple", ArrowDataType::Int32, true)]),
            true,
        );
        let expected = vec![f.clone()];
```